src/ZODB/DB.py

```diff
@@ -438,12 +444,10 @@ def __init__(self, storage,
                 DeprecationWarning, 2)
             storage.tpc_vote = lambda *args: None
 
-        if IMVCCStorage.providedBy(storage):
-            temp_storage = storage.new_instance()
-        else:
-            temp_storage = storage
+        temp_storage = self._mvcc_storage.new_instance()
```

Comment:
Yes, except it's more than just a factory. I'd like to clean this up. I'd rather not do it right now.